### ISSUE TITLE
SA1010 incorrectly reported on opening bracket in C# 11 list pattern

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/StyleCopAnalyzers#3503]
```cs
// ⚠️ SA1010 Opening square brackets should not be preceded by a space
//                 ↓
if (new[] { 1 } is [1])
{
}
```

SDK 6.0.200 or newer required. Example csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>preview</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

**Comment by bjornhellander** [comment]
@jnm2, can you point to a large public repository that uses patterns extensively and also uses stylecop? Would be good for testing, to see that at least all existing problems in that repo gets fixed before releasing the next beta.

**Comment by jnm2** [comment]
@bjornhellander Sorry, not off the top of my head.

**Comment by alexrp** [comment]
FWIW, I use patterns a fair amount in some of [my projects](https://github.com/orgs/vezel-dev/repositories) and have quite a few StyleCop rules disabled due to bugs (some relating to patterns, some not).

**Comment by bjornhellander** [comment]
@jnm2 No problem!
@alexrp Excellent. I will have a look. Thanks!

**Comment by AArnott** [comment]
It shows up in pattern matching within `switch` expressions as well:

```cs
    internal static TransparentAddress? TryParse(ReadOnlySpan<char> address)
    {
        if (address.Length < 2)
        {
            return null;
        }

        Span<byte> decoded = stackalloc byte[DecodedLength];
        Base58Check.Decode(address, decoded);
        return decoded[..2] switch
        {
            [0x1c, 0xb8] or [0x1d, 0x25] => new TransparentP2PKHAddress(address),
            [0x1c, 0xbd] or [0x1C, 0xBA] => new TransparentP2SHAddress(address),
            _ => null,
        };
    }
```

In the above snippet, `SA1010` appears on the `[` of `[0x1d` and `[0x1C`.

**Comment by bjornhellander** [comment]
Thanks for the example, @AArnott. The existing pull request should handle this case as well, if it gets merged.
